mriqc/utils.py

```diff
@@ -58,10 +58,15 @@ def gather_bids_data(dataset_folder, subject_inclusion=None, scan_type=None):
         if get_func:
             functional_scans = sorted(glob(op.join(
                 dataset_folder, subject_id, "func",
-                "%s_*bold.nii.gz" % subject_id, )))
+                "%s_*bold*.nii.gz" % subject_id, )))
 
         for i, scan in enumerate(functional_scans):
-            scid = 'func_%d' % (i+1)
+            taskinfo = re.search(r'.+?task-(([^_]+)(_run-([0-9]+))?)_bold.nii.gz$', scan)
+            if taskinfo == None:
```

Comment:
@oesteban thanks! I updated that line and also make a slight improvement to deal with non .gz image files.